extensions/mssql/src/objectManagement/ui/databaseDialog.ts

```diff
@@ -207,6 +243,20 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
 		if (collationNames?.length > 0 && !collationNames.some(name => name.toLowerCase() === this.objectInfo.collationName?.toLowerCase())) {
 			errors.push(localizedConstants.CollationNotValidError(this.objectInfo.collationName ?? ''));
 		}
+
+		// Validate Rows Filegroup names
+		if (this.objectInfo.filegroups?.length > 0) {
+			let seenFilegroups = new Set(this.objectInfo.filegroups.map(function (item) { return item.name }));
+			if (seenFilegroups.size !== this.objectInfo.filegroups.length) {
+				this.objectInfo.filegroups.forEach((item) => {
+					if (seenFilegroups.has(item.name)) {
+						seenFilegroups.delete(item.name);
```

Comment:
Since the set is a record of what filegroups we've seen so far, wouldn't it make more sense to add them to the set as we encounter them and throw the error when we hit a duplicate?

Reply:
We can do that too! But it will be the same as my above case 1(doing map on every change) it will unnecessary enormous number of iteration for no filegroups changes too.

but here, we are deciding whether to iterate the objectInfo or not based on seenSet.size with objectInfo.length, an extra layer of check to avoid unnecessary iterations.

Reply:
Actually you are right! we can do that, modified accordingly